src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -264,15 +270,48 @@ def set_consent_preferences(
                 raise HTTPException(
                     status_code=HTTP_400_BAD_REQUEST, detail=Pii(str(exc))
                 )
+    identity = Identity()
+    setattr(
+        identity,
+        provided_identity.field_name.value,  # type:ignore[attr-defined]
+        provided_identity.encrypted_value["value"],  # type:ignore[index]
+    )
 
-    return _prepare_consent_preferences(db, provided_identity)
+    consent_preferences: ConsentPreferences = _prepare_consent_preferences(
+        db, provided_identity
+    )
+
+    privacy_request_results: BulkPostPrivacyRequests = create_privacy_request_func(
+        db=db,
+        data=[
+            PrivacyRequestCreate(
+                identity=identity,
+                policy_key=DEFAULT_CONSENT_POLICY,
+                consent_preferences=[
+                    consent.dict() for consent in consent_preferences.consent if consent
+                ],
+            )
+        ],
+        authenticated=True,
+    )
+
+    if privacy_request_results.failed or not privacy_request_results.succeeded:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=privacy_request_results.failed[0].message,
+        )
+
+    consent_request.privacy_request_id = privacy_request_results.succeeded[0].id
+    consent_request.save(db=db)
+
+    return consent_preferences
```

Comment:
Currently returning the same consent preferences response as before, but now a privacy request has also been created.  We might return a new response with a combination of both resources.

Reply:
Updating the API response would fall under this PR https://github.com/ethyca/fides/issues/2142 as we decide what the PrivacyCenter should display after Consent preferences are saved in Fides and then a follow-up Privacy Request is queued.